### PR TITLE
Rough work on getting advanced trace features in to the UI.

### PR DESCRIPTION
**Pull request by AWoloszyn**
Rough-out adding advanced trace options in the UI.

### CODE REVIEW

File: gapic/src/main/com/google/gapid/views/TracerDialog.java [hunk 1]
```diff
@@ -458,6 +461,41 @@ protected void configureDialog(DirectoryDialog dialog) {
...
+        observeFramebuffers = null;
+        ofl = null;
+        if (models.settings.traceAdvanced) {
+          ofl = createLabel(this, "");
+          observeFramebuffers = withLayoutData(
+              createCheckbox(this, "Observe Framebuffers", models.settings.traceObserveFramebuffers),
+              new GridData(SWT.FILL, SWT.FILL, true, false));
+        }
+
+        Listener advancedListener = e -> {
```
Comment (AWoloszyn) [inline gapic/src/main/com/google/gapid/views/TracerDialog.java]: This is definitely not what we want. But I dont know swt enough to make anything cleaner.

File: gapic/src/main/com/google/gapid/views/TracerDialog.java [hunk 2]
```diff
@@ -458,6 +461,41 @@ protected void configureDialog(DirectoryDialog dialog) {
...
+          } else {
+            ofl.dispose();
+            ofl = null;
+            observeFramebuffers.dispose();
+            observeFramebuffers = null;
+          }
+          layout(true, true);
+          parent.layout(true, true);
+
+          parent.getParent().getParent().layout(true, true);
```
Comment (AWoloszyn) [inline gapic/src/main/com/google/gapid/views/TracerDialog.java]: This is obviously wrong, any suggestions.

File: gapic/src/main/com/google/gapid/views/TracerDialog.java (hunk 1 above)
Comment (stellama0208) [reply gapic/src/main/com/google/gapid/views/TracerDialog.java]: The SWT widgets' method .setVisible(boolean) might be helpful. You may use it to replace creating and disposing.

Comment (stellama0208) [reply gapic/src/main/com/google/gapid/views/TracerDialog.java]: It's worth noting that after .setVisible(false), the widget will still occupy a space, it just goes transparent. So if there's other setting checkboxes following in the future, there would be a weird blank line in the middle. I think maybe you can:
1.Make the dependent checkbox staying on the same line, and use .setVisible(false) when unwanted. No blank line.
2.Give the dependent checkbox some indent, and use .setEnabled(false) when unwanted. The checkbox will go grey and the UX is still straightforward.

**Comment by AWoloszyn** [comment]
I will close this, we need a proper UI for this.

**Pull request closed without merging**